src/freetype_manager.cc

```diff
@@ -0,0 +1,596 @@
+        iconv(cd, &input, &charInPutLen, &tmp, &output_size);
+        iconv_close(cd);
+
+        return (1024 - output_size) / 4;
+    }
+}
+
+static int UtoL(const char* input, size_t charInPutLen)
+{
+    size_t output_size = 1024;
+    iconv_t cd = iconv_open(current->encoding, "UCS-4-INTERNAL");
+    char* tmp = (char*)output;
+    iconv(cd, &input, &charInPutLen, &tmp, &output_size);
+    iconv_close(cd);
+
+    return (1024 - output_size);
+}
+
+static FtFontGlyph GetFtFontGlyph(uint32_t unicode)
+{
+    if (current->map.count(unicode) > 0) {
+        return current->map[unicode];
+    } else {
+        if (unicode == '\x95') {
+            current->map[unicode].left = 0;
+            current->map[unicode].top = current->maxHeight / 2;
+            current->map[unicode].width = knobWidth;
+            current->map[unicode].rows = knobHeight;
+            current->map[unicode].buffer = knobDump;
+        } else {
+            FT_Load_Glyph(current->face, FT_Get_Char_Index(current->face, unicode), FT_LOAD_DEFAULT | FT_LOAD_NO_BITMAP);
+            FT_Render_Glyph(current->face->glyph, FT_RENDER_MODE_NORMAL);
+
+            current->map[unicode].left = current->face->glyph->bitmap_left;
+            current->map[unicode].top = current->face->glyph->bitmap_top;
+            current->map[unicode].width = current->face->glyph->bitmap.width;
+            current->map[unicode].rows = current->face->glyph->bitmap.rows;
+
+            int count = current->face->glyph->bitmap.width * current->face->glyph->bitmap.rows;
+
+            if (count > 0) {
+                current->map[unicode].buffer = (unsigned char*)internal_malloc_safe(count, __FILE__, __LINE__); // FONTMGR.C, 259
+                memcpy(current->map[unicode].buffer, current->face->glyph->bitmap.buffer, count);
+            } else {
+                current->map[unicode].buffer = NULL;
+            }
+        }
+
+        return current->map[unicode];
+    }
+}
+
+// 0x441C80
+int FtFontsInit()
+{
+    int currentFont = -1;
+
+    for (int font = 0; font < FT_FONT_MAX; font++) {
+        if (FtFontLoad(font) == -1) {
+            gFtFontDescriptors[font].maxHeight = 0;
+            gFtFontDescriptors[font].filebuffer = NULL;
+        } else {
+            ++gFtFontsLength;
+
+            if (currentFont == -1) {
+                currentFont = font;
+            }
+
+            gFtFontManager.maxFont = gFtFontsLength + 100;
+        }
+    }
+
+    if (currentFont == -1) {
+        return -1;
+    }
+
+    gFtFontsInitialized = true;
+
+    FtFontSetCurrentImpl(currentFont + 100);
+
+    return 0;
+}
+
+// 0x441CEC
+void FtFontsExit()
+{
+    for (int font = 0; font < FT_FONT_MAX; font++) {
+        if (gFtFontDescriptors[font].filebuffer != NULL) {
+            internal_free_safe(gFtFontDescriptors[font].filebuffer, __FILE__, __LINE__); // FONTMGR.C, 124
+        }
+    }
+    //TODO: clean up
+}
+
+// 0x441D20
+static int FtFontLoad(int font_index)
+{
+    char string[56];
```

Comment:
In proper ttf fonts the horizontal bearings and the advance are all encoded in the glyph metrics. These manually set up, resolution specific settings from the ini files can all be calculated for the requested em size on the fly. Both maxWidth and maxHeight could be iteratively found for a particular em size without rendering using `FT_Get_First_Char` and `FT_Get_Next_Char` in a while loop at font load time.